### PR TITLE
415 Offline

### PR DESCRIPTION
**Pull request by Skagevang**
Solves #415 when using yarn start. Don't work for yarn build + serve. And I don't know why... Need to read more about the plugin. Only testet on Chrome and Edge.

This PR use [offline-plugin](https://github.com/NekR/offline-plugin) that is used for webpack-project. I think we could do some more tweaking here. Since yarn build+serve don't work I haven't tested with yarn build:prod.

Also did some changes to the manifest, so you could download the page as an app on your computer or mobile. Need to add more icons, and make a banner so people could download it, and not through the console. But that should be in a later PR

### CODE REVIEW

File: src/assets/favicons/generated/site.webmanifest [hunk 1]
```diff
@@ -1,14 +1,16 @@
 {
-    "name": "",
-    "short_name": "",
+    "name": "Kodeklubben Oppgaver",
```
Comment (NorwegianKiwi) [inline src/assets/favicons/generated/site.webmanifest]: Kanskje legge til en kommentar i `src/assets/favicons/README.md` om at du har endret en generert fil?

File: src/index.js [hunk 2]
```diff
@@ -3,6 +3,9 @@ import './polyfills';
 import renderDynamic from './renderDynamic';
 import renderStatic from './renderStatic';
 
+import * as OfflinePluginRuntime from 'offline-plugin/runtime';
+OfflinePluginRuntime.install();
```
Comment (NorwegianKiwi) [inline src/index.js]: Hvis du flytter `install`-linja innenfor `if`-en rett over `renderDynamic()`, så funker `yarn build` + `yarn serve`.

**Comment by NorwegianKiwi** [comment]
Jeg gikk i console og prøvde å slette all cache for service workers, og da jeg da prøvde å reloade siden fikk jeg `Error: Wrong response status` i console.

File: src/index.js (hunk 2 above)
Comment (Skagevang) [reply src/index.js]: Ja har prøvd litt forskjellig her, men med den måten så tror jeg ikke service worker fungerer

File: src/assets/favicons/generated/site.webmanifest (hunk 1 above)
Comment (Skagevang) [reply src/assets/favicons/generated/site.webmanifest]: nja kanskje, men en manifest fil er ganske lurt iforhold til pwa uansett
https://developer.mozilla.org/en-US/docs/Web/Manifest

File: src/index.js (hunk 2 above)
Comment (NorwegianKiwi) [reply src/index.js]: Hvorfor ikke?

**Comment by Skagevang** [comment]
Closer denne fordi jeg kommer nok heller til å bruke denne pluginen: https://github.com/oliviertassinari/serviceworker-webpack-plugin og skrive min egen service worker i sommer. I tillegg må jeg huske at den ikke skal registrere service worker under localhost. S

**Pull request closed without merging**